lib/interaction_machine.js

```diff
@@ -670,6 +672,7 @@ var InteractionMachine = Eventable.extend(function(self, api, app) {
                 return self.state.continue_session();
             })
             .then(function(continue_session) {
+                opts.helper_metadata = self.state.helper_metadata;
```

Comment:
I'm wondering whether we shouldn't do this as:
```javascript
.then(function() {
    return Q.all([self.state.continue_session(), self.state.helper_metadata()]);
})
.then(function(continue_session, helper_metadata) {
...
})
```
so that we can do on the state:
```javascript
self.helper_metadata = utils.functor(opts.helper_metadata);
```
so helper metadata can be a constant, a function or a function that returns a promise.

Aside: chaining the promises rather than using ``Q.all`` would guarantee the ordering a bit more which might be useful, but maybe it's not worth the slight clumsiness of not doing them in parallel.